corehq/apps/sms/management/commands/run_sms_queue.py

```diff
@@ -53,11 +66,27 @@ def enqueue(self, queued_sms):
     def handle(self, **options):
         while True:
             try:
-                self.create_tasks()
+                if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS and not can_send_sms_now():
+                    sleep(self._time_till_next_window())
```

Comment:
sleeping for a very long time is not a good idea, rather sleep for a few seconds at a time

Reply:
thanks for raising that, is it just something that should not be done? Since we already know till what point we can't process SMS, i tried to delay retry till then.

Reply:
My gut says it's not a great idea but I couldn't find any actual documentation to back that up.

Reply:
I am okay with keeping this to a shorter duration, its not expensive to recheck it anyway. Let me know if you have a suggestion on how much delay is okay